common/log.c

```diff
@@ -634,6 +694,149 @@ log_end(void)
     return ret;
 }
 
+/*****************************************************************************/
+/* produce a hex dump */
+enum logReturns
+log_hexdump_with_location(const char *function_name, 
+                          const char *file_name, 
+                          const int line_number, 
+                          const enum logLevels log_level, 
+                          const char *message, 
+                          const char *src, 
+                          int len)
+{
+    unsigned char *line;
+    int i;
+    int dump_number_lines;
+    int dump_line_length;
+    int dump_header_length;
+    int dump_length;
+    int dump_offset;
+    int thisline;
+    int offset;
+    char *dump_buffer;
+    enum logReturns rv;
+    bool_t override_destination_level = 0;
+
+    override_destination_level = internal_log_location_overrides_level(function_name, 
+                                                                       file_name, 
+                                                                       log_level);
+    if(!internal_log_is_enabled_for_level(log_level, override_destination_level))
+    {
+        return LOG_STARTUP_OK;
+    }
+
+    dump_line_length = 4 + 3             /* = 4 offset + 3 space */
+                       + ((2 + 1) * 16)  /* + (2 hex + 1 space) per byte for 16 bytes */
```

Comment:
Suggest making the `16` a macro as it's used a lot.

Reply:
agreed, fixed